packages/core/content-type-builder/server/services/schema-builder/utils/convert-custom-field-type.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+/**
+ * @description
+ * Sets attribute.type to customField
+ * @param {object} attributes Attributes found on content-type or component
+ */
+const convertCustomFieldType = attributes => {
+  Object.values(attributes).forEach(attribute => {
+    if (attribute.customField) {
+      // Use the custom field uid sent from the admin to get its equivalent on the server
+      // The getter will throw an error if the custom field is not found
+      const customField = strapi.container.get('custom-fields').get(attribute.customField);
+
+      if (customField.type !== attribute.type) {
```

Comment:
I'm not sure I understand why the frontend isn't sending { type: 'customField'} directly ? Looks like we do transforms we could avoid 🤔  We would just convert before sending the schema to the frontend instead of after receiving from the frontend. I thought this was what we defined

Reply:
ping @remidej  Do you have some info on this? I'm not really comfortable going live with this approach

Reply:
I'll look into it today

Reply:
I'll look into it today